Tech/ci examples overhaul

ndelangen opened the pull request.

gaetanmaisse committed:
feat: add e2e tests for angular

ndelangen committed:
CHANGE structure of transform scripts

ndelangen committed:
USE verdaccio in CI

ndelangen committed:
FIX program import

ndelangen committed:
REMOVE checks

ndelangen committed:
CHANGE location of verdaccio storage

ndelangen committed:
ADD port to registry script && ADD a small test to check if local registry is indeed working in CI

ndelangen committed:
CHANGE to use verdaccio node-api

ndelangen committed:
FIX wait-on

ndelangen committed:
Parallel publish

ndelangen committed:
FIX

ndelangen committed:
FIX port

ndelangen committed:
ADD comment

ndelangen committed:
FIX port some more

ndelangen committed:
FIX

ndelangen committed:
force registry

ndelangen committed:
IMPROVE caching

ndelangen committed:
ADD more set config registry, argh

ndelangen committed:
CLEANUP

ndelangen committed:
CLEANUP

ndelangen committed:
FIX eslint

ndelangen committed:
CLEANUP

ndelangen committed:
CLEANUP

ndelangen committed:
Merge branch 'next' into tech/ci-examples-overhaul

ndelangen committed:
CLEANUP

ndelangen committed:
FIX transformers && CLEANUP

ndelangen committed:
CLEANUP

ndelangen committed:
CHANGE run-chromatics script to use esm

ndelangen committed:
CHANGE to use esm for repo-dirst-check

ndelangen committed:
RENAME

ndelangen committed:
RESTRUCTURE script utils a bit

ndelangen committed:
REMOVE devDependencies of cli

ndelangen committed:
FIX cli-util location

ndelangen committed:
FIX paths

ndelangen committed:
FIX linting

ndelangen committed:
FIX linting

ndelangen committed:
CHANGE use of devDependencies to a versions.json file

This makes the monorepo lighter, since we don't depend on storybook/react-native anymore

ndelangen committed:
CLEANUP lockfile

ndelangen committed:
UPGRADES

ndelangen committed:
FIX missing devDependency

juanpicado commented on package.json:
```diff
@@ -234,9 +234,11 @@
     "cypress": "^4.0.2",
-    "puppeteer": "^2.1.1"
+    "puppeteer": "^3.0.0",
+    "verdaccio": "^4.5.1",
```
Side note `v^4.5.1` is only compatible with Node `v10` or higher.  I see a few lines below the engine node constrains.

ndelangen replied:
Storybook only support node v10 and up.

gaetanmaisse committed:
fix: add `@storybook/csf` to the list of packages proxied to npmjs by verdaccio

gaetanmaisse committed:
fix: add `versions.json` to packaged files of `@storybook/cli`

shilman commented on lib/cli/versions.json:
```diff
@@ -0,0 +1,16 @@
+{
+  "@storybook/angular": "6.0.0-alpha.37",
```
What's the idea here? Can you explain where this going?

gaetanmaisse replied:
I think the goal is to use this file instead of `devDependencies` field of `package.json` of `@storybook/cli`, see here: https://github.com/storybookjs/storybook/pull/10447/files#diff-82f09434813e71512ceb5d2a5662c331R12

ndelangen replied:
correct!

ndelangen committed:
Merge branch 'next' into tech/ci-examples-overhaul

# Conflicts:
#	lib/cli/package.json
#	package.json
#	yarn.lock

ndelangen committed:
Merge branch 'init-e2e' into tech/ci-examples-overhaul

ndelangen committed:
Merge branch 'next' into tech/ci-examples-overhaul

# Conflicts:
#	lib/cli/package.json

gaetanmaisse committed:
Add some TODO for the next steps after a review with @ndelangen

ndelangen committed:
WIP

ndelangen committed:
CHANGE serve to TS

ndelangen committed:
FIX ts issues

ndelangen committed:
ADD types

gaetanmaisse committed:
Add directory created during e2e tests in gitignore

ndelangen committed:
CLEANUP

ndelangen committed:
Merge branch 'tech/ci-examples-overhaul' of github.com:storybookjs/storybook into tech/ci-examples-overhaul

yannbf commented on lib/cli/versions.json:
```diff
@@ -0,0 +1,16 @@
+{
+  "@storybook/angular": "6.0.0-alpha.43",
```
will these versions always be the same and match with all other @storybook/*?

ndelangen replied:
yes, it's auto-generated in a prepare script!

ndelangen committed:
bootstrap angular in cwd

ndelangen committed:
FIX linting

ndelangen committed:
temp rename root node_modules

gaetanmaisse committed:
test: bootstrap E2E tests for Yarn 2

tooppaaa committed:
add vue script

yannbf committed:
test: fix paths for cypress and static folder

ndelangen committed:
REMOVE adding a user

yannbf committed:
fix(e2e): remove unnecessary quotes on getUrl

yannbf committed:
chore: upgrade cypress to 4.4.1

yannbf committed:
chore: change e2e script

yannbf committed:
test(e2e): improve helpers

yannbf committed:
test(e2e): add basic flow tests

yannbf committed:
test(e2e): run cypress with custom config

yannbf committed:
Merge branch 'tech/ci-examples-overhaul' of https://github.com/storybookjs/storybook into tech/ci-examples-overhaul

yannbf committed:
chore(e2e): add cypress custom typings

yannbf committed:
chore(e2e): add cypress console types

yannbf committed:
chore(e2e): disable jest/expect rule for cypress

ndelangen committed:
CHANGE run cypress tests in cypress docker container

ndelangen committed:
REMOVE unneeded dependency

ndelangen committed:
FIX lockfile & linting

ndelangen committed:
CHANGE to use cypress/node10

ndelangen committed:
ENABLE the script features

ndelangen committed:
ADD prompts questions for cleanup

ndelangen committed:
Merge branch 'next' into tech/ci-examples-overhaul

# Conflicts:
#	lib/cli/package.json
#	yarn.lock

ndelangen committed:
CHANGE to use siblingDirectory rather then rename hackery

ndelangen committed:
hopefully fix the bootstrapping of angular on CI && better logging && CLEANUP

ndelangen committed:
force upgrade yarn

ndelangen committed:
UPGRADE to latest version of docker image for cypress

ndelangen committed:
CHANGE to use cypress/included:4.5.0

ndelangen committed:
ADD cypress command switcher && CLEANUP && upgrade cypress

ndelangen committed:
FIX for some firefox flake

ndelangen committed:
increase timeouts of cypress

ndelangen committed:
share install & build step in CI && ADD artifacts

tooppaaa committed:
industrialise script

ndelangen committed:
FIX process not exiting on CI

ndelangen committed:
Merge branch 'next' into tech/ci-examples-overhaul

# Conflicts:
#	lib/cli/package.json

ndelangen committed:
Merge branch 'next' into tech/ci-examples-overhaul

# Conflicts:
#	package.json
#	yarn.lock

tooppaaa committed:
rename welcome

tooppaaa committed:
split test

tooppaaa committed:
windows friendly

tooppaaa committed:
e2e from config

tooppaaa committed:
add svelte config

tooppaaa committed:
fix yarn.lock

tooppaaa committed:
make ids 6.0 friendly

tooppaaa committed:
add preact

tooppaaa committed:
add ember and all placeholders

tooppaaa committed:
add ember strories

tooppaaa committed:
Vue all green

gaetanmaisse committed:
fix(actions): fix import of `uuid`

As of 8d35431027a0e849f8fcfe4f449db82f30136b77 `@storybook/addon-actions` is using version 8.0.0 of `uuid`
However, a breaking change has been missed during the upgrade:
https://github.com/uuidjs/uuid/blob/master/CHANGELOG.md#800-2020-04-29

TLDR;
> Deep requiring specific algorithms of this library like require('uuid/v4'), which has been deprecated in uuid@7, is no longer supported.

tooppaaa committed:
attempt to add marko

tooppaaa committed:
add cra scripts

tooppaaa committed:
Merge remote-tracking branch 'origin/next' into tech/ci-examples-overhaul

tooppaaa committed:
cleanup autodetect

tooppaaa committed:
add mithril

tooppaaa committed:
Revert "cleanup autodetect"

This reverts commit cf0968452a162be7051c3a4f18eb9f2164742280.

tooppaaa committed:
add html

tooppaaa committed:
add web_components

tooppaaa committed:
fix html & preact

tooppaaa committed:
add react

tooppaaa committed:
add rax, react, sfc-vue, webpack-react

tooppaaa committed:
fix react buttons

tooppaaa committed:
fix vue sfc

gaetanmaisse committed:
chore: add some info about work to do to make Marko E2E work

gaetanmaisse committed:
chore: add `additionalDeps` property to avoid installing react and react-dom in tests of all frameworks

gaetanmaisse committed:
chore: add E2E test for CRA using Yarn 2

gaetanmaisse committed:
chore: rename E2E tests step in Circle CI

gaetanmaisse committed:
chore: add a job dedicated to Yarn E2E tests in Circle CI

gaetanmaisse committed:
chore: uncomment CRA E2E tests

gaetanmaisse committed:
chore: remove Yarn 2 E2E test files as it is now handled as all other tests

tooppaaa committed:
Merge remote-tracking branch 'origin/next' into tech/ci-examples-overhaul

tooppaaa committed:
test parallelism

tooppaaa committed:
add private to aurelia example

tooppaaa committed:
paralell CI

tooppaaa committed:
increase parallel & try reporter

tooppaaa committed:
remove test results (not working)

tooppaaa committed:
fix cra

tooppaaa committed:
exclude node-logger

tooppaaa committed:
ignore node logger proxy

tooppaaa committed:
remove proxy

yannbf committed:
refactor: export object with framework configs

yannbf committed:
fix(run-e2e): move catch to the bottom

yannbf committed:
Merge branch 'tech/ci-examples-overhaul' of https://github.com/storybookjs/storybook into tech/ci-examples-overhaul

yannbf committed:
Revert "refactor: export object with framework configs"

This reverts commit a04dd4fe7c8f3319626f49eade25f53510462126.

tooppaaa committed:
Merge remote-tracking branch 'origin/next' into tech/ci-examples-overhaul
